config/config_handler.go

```diff
@@ -158,6 +137,43 @@ func (ch *pdConfHandler) GetConfig() *Config {
 	return ch.curConf.Load().(*Config)
 }
 
+func (ch *pdConfHandler) SetConfig(conf *Config) error {
+	return errors.New("PDConfHandler only support to update the config from PD whereas forbid to modify it locally")
+}
+
+func (ch *pdConfHandler) register() {
+	// register to PD and get the new default config.
+	// see https://github.com/pingcap/tidb/pull/13660 for more details.
+	// suppose port and security config items cannot be change online.
+	confContent, err := encodeConfig(ch.curConf.Load().(*Config))
+	if err != nil {
+		logutil.Logger(context.Background()).Warn("encode config error when registering", zap.Error(err))
+		return
+	}
+
+	status, version, conf, err := ch.pdConfCli.Create(context.Background(), new(configpb.Version), tidbComponentName, ch.id, confContent)
+	if err != nil {
+		logutil.Logger(context.Background()).Warn("RPC to PD error when registering", zap.Error(err))
+		return
+	} else if status.Code != configpb.StatusCode_OK && status.Code != configpb.StatusCode_WRONG_VERSION {
+		logutil.Logger(context.Background()).Warn("invalid status when registering", zap.String("code", status.Code.String()), zap.String("errmsg", status.Message))
+		return
+	}
+
+	newConf, err := decodeConfig(conf)
+	if err != nil {
+		logutil.Logger(context.Background()).Warn("decode config error when registering", zap.Error(err))
+		return
+	} else if err := newConf.Valid(); err != nil {
+		logutil.Logger(context.Background()).Warn("invalid remote config when registering", zap.Error(err))
+		return
+	}
+
+	ch.registered = true
+	ch.version = version
+	logutil.Logger(context.Background()).Info("register the config to PD successful", zap.String("new_config", conf))
```

Comment:
1. s/ successful/ successfully
2. s/ conf/ newConf ?
3. Should we log the `version`